src/sentry/eventstream/kafka/backend.py

```diff
@@ -248,6 +276,20 @@ def commit_offsets():
                 )
                 commit(offsets_to_commit)
 
+        def process_message(kafka_message: Message) -> None:
+            task_kwargs = self._get_task_kwargs(kafka_message)
+            if not task_kwargs:
+                return
+
+            self.compute_metrics(kafka_message.partition(), task_kwargs)
+            self._pending_futures.append(
+                executor.submit(self._dispatch_post_process_group_task, **task_kwargs)
+            )
+
+        def collect_results() -> None:
+            wait(self._pending_futures)
```

Comment:
Is this where exceptions are thrown? It seems like since you split the task args and dispatch function the error handling is not the same as it was before.

Reply:
Good catch. I changed it to futures.as_completed and check for exceptions which get thrown while running the task.